src/k2/transport/Payload.h

```diff
@@ -165,7 +165,10 @@ class Payload {
     Payload shareRegion(size_t startOffset, size_t nbytes);
 
     // Creates a new payload as a copy of this payload. The underlying data is copied over to the new payload
-    Payload copy(BinaryAllocatorFunctor dest_allocator=DefaultAllocator);
+    Payload copy(BinaryAllocator dest_allocator);
```

Comment:
since we're keeping this allocator, i think you should receive by &&

Reply:
done